### ISSUE TITLE
Certs plugin - nginx.ssl.conf.template is not used for HTTPS urls

### ISSUE THREAD

**Issue by darklow** [progrium/dokku#1516]
Dokku version 0.4.1 / Ubuntu 14.04
I added certificate using new certs plugin.
Now `VHOST` has following entries:
```
mydomain.com
app.mydomain.com
```

URLS file:
```
https://app.mydomain.com
http://mydomain.com
```

Rebuilding:
```
 ssh dokku@jedi nginx:build-config mydomain
-----> Configuring SSL for app.mydomain.com...
-----> Configuring mydomain.com...
-----> Creating https nginx.conf
-----> Running nginx-pre-reload
       Reloading nginx
```

But the final `nginx.conf` contains both parts from `nginx.conf.template`:
```
server {
  listen      [::]:80;
  listen      80;
  server_name app.mydomain.com;
...
}
server {
  listen      [::]:80;
  listen      80;
  server_name mydomain.com;
...
}
```

So looks like there is a bug and script doesn't use `nginx.ssl.conf.template` for SSL subdomain part.

**Comment by darklow** [comment]
More details: Apparently it happens only when using custom `nginx.conf.template`.
Even though i copied in same directory also `nginx.ssl.conf.template` it is not used and for both part SSL and NON-SSL same template is used.

**Comment by josegonzalez** [comment]
I'll fix the issue in exchange for a djangosuit OEM integrator license :P

We appear to be setting the template to your custom one [here](https://github.com/progrium/dokku/blob/master/plugins/nginx-vhosts/functions#L157). Note that we only use the custom `nginx.conf.template`, regardless of whether there is a custom `nginx.ssl.conf.template` file or not. We may want to *also* support a custom ssl cert, or at the very least make it absolutely clear as to what happens with custom templates.

I'm a bit confused by your last comment. Are you saying that no custom `nginx.conf.template` is being used, or just that your custom `nginx.ssl.conf.template` isn't being picked up?

**Comment by darklow** [comment]
@josegonzalez i can offer you discount on OEM :)
I just checked out functions file and noticed that custom `nginx.ssl.conf.template` is not even handled.
But it was easy fixable with a few lines of code, i can offer a pull request with a fix.

**Comment by josegonzalez** [comment]
Cool, feel free to pr. Would be good to also update docs :)

**Comment by josegonzalez** [comment]
@Flink is this closed by #1530 ?

**Comment by darklow** [comment]
@josegonzalez It fixes partially but it still doesn't allow to use custom templates for SSL with `nginx.ssl.conf.template`
Give me a few days, i'll submit a PR

**Comment by josegonzalez** [comment]
Does `nginx.conf.template` not work for SSL?

**Comment by darklow** [comment]
@josegonzalez that was the problem `nginx.conf.template` was used for both parts SSL and NON-SSL. This bug @Flink just fixed in #1530 
However in code there are references to `nginx.ssl.conf.template` but it is actually never used.
